playbooks/openstack/openshift-cluster/build_image.yml

```diff
@@ -0,0 +1,177 @@
+---
+- name: Verify image build prerequisites
+  hosts: localhost
+  connection: local
+  gather_facts: no
+  tasks:
+  - name: Require openshift_openstack_build_base_image
+    fail:
+      msg: "A base image name is required for image building. Please ensure `openshift_openstack_build_base_image` is defined."
+    when: openshift_openstack_build_base_image is undefined
+
+  - name: Require openshift_openstack_default_image_name
+    fail:
+      msg: >
+        You must specify the name the new image will be saved to Glance as.
+        Please make sure `openshift_openstack_default_image_name` is defined.
+    when:
+    - openshift_openstack_default_image_name is not defined
+
+  - name: Get the target image facts
+    os_image_facts:
+      image: "{{ openshift_openstack_default_image_name }}"
+    register: image_check
+
+  - name: Verify that the resulting image does not exist
+    fail:
+      msg: >
+        The image "{{ openshift_openstack_default_image_name }}" specified in
+        `openshift_openstack_default_image_name` exists already.
+        Please choose a different name or delete it.
+    when: image_check.ansible_facts.openstack_image
+
+  - set_fact:
+      build_prefix: "{{ openshift_openstack_clusterid|default('build') }}.{{ openshift_openstack_public_dns_domain|default('example.com') }}-build"
+
+
+- name: Launch image build instance
+  hosts: localhost
+  connection: local
+  gather_facts: yes
+  tasks:
+  # NOTE: we create a temporary network, subnet, router and security groups
+  # to have a known environment to launch the image build VM in.
+  # They get deleted once we save the image.
+  # TODO(shadower): allow specifying an existing subnet etc. instead.
+  - name: Create a network
+    os_network:
+      name: "{{ build_prefix }}-network"
+    register: network
+
+  - name: Create a subnet
+    os_subnet:
+      name: "{{ build_prefix }}-subnet"
+      network_name: "{{ network.network.name }}"
+      cidr: 192.168.23.0/24
```

Comment:
this really should be configurable

Reply:
Yep, will be fixed in the next revision.